### PR TITLE
Added parsing from String for decoding Int64

### PR DESCRIPTION
**Pull request by nvh**
Implementation of solution for https://github.com/thoughtbot/Argo/issues/300

### CODE REVIEW

File: Argo/Types/StandardTypes.swift [hunk 1]
```diff
@@ -22,6 +22,12 @@ extension Int64: Decodable {
   public static func decode(j: JSON) -> Decoded<Int64> {
     switch j {
     case let .Number(n): return pure(n.longLongValue)
+    case let .String(s):
+      if let i = Int64(s) {
+        return pure(i)
+      } else {
+        return .typeMismatch("Int64", actual: s)
```
Comment (gfontenot) [inline Argo/Types/StandardTypes.swift]: Possible to use `fallthrough` here?

**Comment by gfontenot** [comment]
I think this is great. Solid idea keeping `Number` support while adding `String` support.

File: Argo/Types/StandardTypes.swift (hunk 1 above)
Comment (nvh) [inline Argo/Types/StandardTypes.swift]: Good point, fixed in e08fed7a288533ce6c80f9a0c80fde6e15ed6dcb

**Comment by nvh** [comment]
I was trying to get it in 1 line, but there is no way to do something like
```swift
if case let .String(s) = j, let i = Int64(s)
```
in a switch statement, right?

File: Argo/Types/StandardTypes.swift [hunk 2]
```diff
@@ -22,6 +22,12 @@ extension Int64: Decodable {
   public static func decode(j: JSON) -> Decoded<Int64> {
     switch j {
     case let .Number(n): return pure(n.longLongValue)
+    case let .String(s):
+      if let i = Int64(s) {
+        return pure(i)
+      } else {
```
Comment (tonyd256) [inline Argo/Types/StandardTypes.swift]: shouldn't need the `else` ... it will return if it succeeds, stoping execution.

Comment (tonyd256) [inline Argo/Types/StandardTypes.swift]: could possibly use a guard?

**Comment by tonyd256** [comment]
I like this. only comment is to condense that case somehow.

File: Argo/Types/StandardTypes.swift (hunk 2 above)
Comment (gfontenot) [inline Argo/Types/StandardTypes.swift]: +1

```swift
guard let i = Int64(s) else { fallthrough }
return pure(i)
```

should work nicely.

Comment (nvh) [inline Argo/Types/StandardTypes.swift]: Fixed in 2e9047c

**Comment by gfontenot** [comment]
Fantastic. Do you mind squashing and rebasing and then I'll merge this in?

**Comment by gfontenot** [comment]
Rebased and merged as e7680fc

Thanks!

**Pull request closed without merging**